### ISSUE TITLE
Parquet scanner doesn't do predicate pushdown for categoricals

### ISSUE THREAD

**Issue by deanm0000** [pola-rs/polars#18868]
### Checks

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Reproducible example

```python
pl.DataFrame([
    pl.Series('a',['a','b','c'], pl.Categorical)]).write_parquet('catpa.parquet',row_group_size=1, use_pyarrow=True)
with pl.Config(verbose=True):
    pl.scan_parquet("catpa.parquet").filter(pl.col('a')=='a').collect()

```

### Log output

```shell
parquet file must be read, statistics not sufficient for predicate.
parquet file must be read, statistics not sufficient for predicate.
parquet file must be read, statistics not sufficient for predicate.
```

### Issue description

This is separate from but highly related to https://github.com/pola-rs/polars/issues/18867. Even when using a file written by pyarrow where the statistics are correct, predicate pushdown doesn't work.

If I try to explicitly make the rhs a Categorical then I simply don't get a verbose message at all so I'm not sure if it's silently working or not working.

```python
with pl.Config(verbose=True):
    print(pl.scan_parquet("catpa.parquet").filter(pl.col('a')==pl.lit('a',pl.Categorical)).collect())
```

Even with a StringCache still no verbosity.

```python
with pl.Config(verbose=True), pl.StringCache():
    print(pl.scan_parquet("catpa.parquet").filter(pl.col('a')==pl.lit('a',pl.Categorical)).collect())
```

### Expected behavior

Partition pruning should work

### Installed versions

<details>

```
--------Version info---------
Polars:              1.7.1
Index type:          UInt32
Platform:            Linux-5.15.153.1-microsoft-standard-WSL2-x86_64-with-glibc2.35
Python:              3.11.9 (main, Apr  6 2024, 17:59:24) [GCC 11.4.0]

----Optional dependencies----
adbc_driver_manager  <not installed>
altair               <not installed>
cloudpickle          <not installed>
connectorx           0.3.2
deltalake            0.18.2
fastexcel            <not installed>
fsspec               2024.3.1
gevent               <not installed>
great_tables         <not installed>
matplotlib           <not installed>
nest_asyncio         1.6.0
numpy                1.26.4
openpyxl             <not installed>
pandas               2.2.2
pyarrow              17.0.0
pydantic             <not installed>
pyiceberg            <not installed>
sqlalchemy           <not installed>
torch                <not installed>
xlsx2csv             <not installed>
xlsxwriter           <not installed>
```

</details>

**Comment by aberres** [comment]
The same happens with enum columns.

**Comment by aberres** [comment]
Some numbers from a real live example:

![CleanShot 2024-11-29 at 15 07 15@2x](https://github.com/user-attachments/assets/bc8a2f6f-ec6a-4cc6-b0c6-7cc55374c9f4)

What I am reading is a ~100Mb Parquet representing a ~30Gb in memory frame. The filter matches no rows.

In the first case, the "Assumption" column has been written as string column, in the second case as categorical column. The data is residing in a Google Cloud bucket in Singapore, this means I/O is costly.

As we see, the missing push-down has a dramatic effect.

**Comment by coastalwhite** [comment]
One thing to note here is that we don't really have any good parquet statistics for `pl.Enum` and `pl.Categorical(ordering=physical)`. This is because parquet only reports a `lexical` minimum and maximum value. We could still do some stuff, but it would be much more complex than what we currently do.

**Comment by kcajf** [comment]
> One thing to note here is that we don't really have any good parquet statistics for `pl.Enum` and `pl.Categorical(ordering=physical)`. This is because parquet only reports a `lexical` minimum and maximum value, while enum and categorical have a physical minimum and maximum. We could still do some stuff, but it would be much more complex than what we currently do.

Is this relevant to predicate pushdown or was it more a comment about https://github.com/pola-rs/polars/issues/18867? With predicate pushdown, most filters will probably be dealing with a set of string literals (e.g. `.filter(col="abc")`) so it's not obvious to me how physical vs. lexical matters when filtering row groups?

**Comment by coastalwhite** [comment]
> With predicate pushdown, most filters will probably be dealing with a set of string literals (e.g. `.filter(col="abc")`) so it's not obvious to me how physical vs. lexical matters when filtering row groups?

Parquet only provides `min`, `max` and `null count` statistics (and `distinct count` but that doesn't matter). Since these follow lexical ordering and enums follow physical ordering, we cannot just do the following reasoning:

The filter is`col(A) == "abc"` so we can skip the row group it if the statistics say `max_of_row_group(column A) < "abc"` or `min_of_row_group(A) > "abc"`.

**Comment by kcajf** [comment]
But isn't evaluating `max_of_row_group(column A) < "abc"` on a parquet row group totally independent of polars / enums?

**Comment by coastalwhite** [comment]
Yes ideally, but we use actual polars expressions with polars data types to evaluate this. So yes theoretically it would be possible if you ensure that the expression knows they need to be treated as strings, but it requires quite a bit of special case handling and fiddling.

**Comment by coastalwhite** [comment]
> Oh, this means for categorical strings it's actually much more efficient to filter your parquet files by them if your parquet data isn't dict encoded?

*Dictionary encoding* and *dictionary types* (i.e. enums/categoricals) in parquet are mostly orthogonal. String columns are still dictionary encoded and enums/categoricals (we don't do this, but we could) could be encoded only by their values. Dictionary encoded strings and dictionary encoded enums have the same in-file representation and use the same decoder.

What is indeed less efficient is to filter over *dictionary types*, but only because we cannot do skipping based on statistics. That may or may not actually be important depending on your use case.
